### ISSUE TITLE
upload: bids - become smarter on where to look for dataset_description.json

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#1072]
Case brought up by @satra in https://github.com/dandi/dandi-cli/issues/1069#issuecomment-1188296871 that in case of 

```
DANDI_DEVEL=1 dandi -l DEBUG --pdb upload --validation skip --allow-any-path sub-MITU01/ses-20211111h11m08s18/micr/sub-MITU01_ses-20211111h11m08s18_sample-11_photo.jpg
```

`dandi` is still traversing entire folder (I guess since it is stuck) instead of just to look for dataset_description.json "upstairs" from the location of that file.  So, for files should look only in/up their folder, for folders - in/down from them.  Marking for `upload` command but I guess it is relevant also for `validate-bids`

**Comment by satra** [comment]
i also posted simultaneously see #1073

**Comment by TheChymera** [comment]
@yarikoptic I don't think this is happening any longer. For one, I don't see any output indicating that from the command you listed, additionally, we now use https://github.com/dandi/dandi-cli/blob/b11835a01380bf7ff70da23ef42e7315a53f2dbe/dandi/utils.py#L406 which only crawls up.

**Comment by yarikoptic** [comment]
if you think it is no longer an issue -- please close. If you could identify commit or PR which fixed that -- would be appreciated.

**Comment by TheChymera** [comment]
Should be fixed as of 122581ad47e3fbf22747b7773ea172a940484464 . Best reopen if this pops up again.
